### PR TITLE
chore: Replace use of flakeheaven with Ruff for pep8-naming

### LINKED ISSUE #8402: Replace use of flakeheaven with Ruff for pep8-naming
* Disable the flakeheaven plugin in `pyproject.toml`
* Migrate to Ruff's `Nxxx` rules
* Remove additional dependency from `pre-commit-config.yaml`

### PR DESCRIPTION
**Pull request by nkclemson**
Closes #8402

 - [ ] squash git commits

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
@nkclemson Ping me when this is ready for a review. Also, don't feel the need to squash commits  before marking this as ready since we do merge squashes anyway :)

**Comment by nkclemson** [comment]
Hi @edgarrmondragon, I think this is complete now. I have added the `"N815" - mixedCase variable in class scope (constants, method names)` exemption, as this is in accordance with [Guido's recommendation](https://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations). Initially, I thought that the lint error I encountered in test_logging_utils.py was N815, but I later realized that those all-caps variable names are inside methods, so I fixed it. let me know if I need to update anything.

File: pyproject.toml [hunk 1]
```diff
@@ -586,6 +579,8 @@ exclude = [
 ignore = [
   "ANN101", # Missing type annotation for `self` in method
   "ANN102", # Missing type annotation for `cls` in class method
+  "N815",   # mixedCase variable in class scope (constants, method names)
```
Comment (edgarrmondragon) [inline pyproject.toml:582]: I don't think this is currently triggered anywhere in the codebase

```suggestion
```

File: pyproject.toml [hunk 2]
```diff
@@ -625,6 +621,9 @@ unfixable = [
 "tests/**" = [
   "S101", # Allow 'assert' in tests
 ]
+"src/meltano/migrations/versions/**" = [
+  "N806", # Allow uppercase variable names
+]
```
Comment (edgarrmondragon) [inline pyproject.toml:626]: I think this is covered by the `exclude` setting?

```suggestion
```

File: pyproject.toml (hunk 1 above)
Comment (nkclemson) [reply pyproject.toml]: yep. This isn't triggered. I am removing this.

File: pyproject.toml (hunk 2 above)
Comment (nkclemson) [reply pyproject.toml]: Yep.

**Review by edgarrmondragon — APPROVED** [review]
Thanks a ton @nkclemson!

**Pull request merged**